p4-16/psa/PSA.mdk

```diff
@@ -309,6 +310,134 @@ Additional supported match_kind types
 ```
 
 
+## Data plane vs. control plane data representations
+
+A PSA data plane implementation that supports the P4 Runtime
+API[^P4RuntimeAPI] is expected to include software called a "P4
+Runtime API agent" that encapsulates details specific to the PSA
+implementation, enabling a P4 Runtime API controller to more easily
+control multiple devices with different PSA implementations.  We will
+refer to these pieces of software as an "agent" and "controller" for
+brevity.
+
+As mentioned in section [#sec-psa-type-definitions], different PSA
+implementations are expected to differ in the minimum number of bits
```

Comment:
Suggestion on rephrasing: different PSA implementations are expected to customize the size of the data types that refer directly to physical entities, such as ports.

Reply:
I have replaced it with this "are expected to customize the size of the data types that refer directly to those objects, i.e. ports, multicast group ids, etc.".  Something just seems a little off to me to refer to multicast group ids and instance ids as physical entities.